src/block/styled-components.js

```diff
@@ -0,0 +1,199 @@
+    $width,
+    $minWidth,
+    $maxWidth,
+    $height,
+    $minHeight,
+    $maxHeight,
+    $overflow,
+    $margin,
+    $marginTop,
+    $marginRight,
+    $marginBottom,
+    $marginLeft,
+    $padding,
+    $paddingTop,
+    $paddingRight,
+    $paddingBottom,
+    $paddingLeft,
+    $flexWrap,
+    $left,
+    $top,
+    $right,
+    $bottom,
+    ...other
+  }: StyledBlockPropsT) => {
+    const {colors, typography, sizing} = $theme;
+
+    let styles = {};
+
+    if ($color !== undefined) {
+      styles.color = colors[$color] || $color;
+    }
+
+    if ($font !== undefined) {
+      const font: Font = typography[$font];
+      styles.fontFamily = font.fontFamily;
+      styles.fontWeight = font.fontWeight;
+      styles.fontSize = font.fontSize;
+      styles.lineHeight = font.lineHeight;
+    }
+
+    if ($alignContent) {
+      styles.alignContent = $alignContent;
+    }
+
+    if ($alignItems) {
+      styles.alignItems = $alignItems;
+    }
+
+    if ($alignSelf) {
+      styles.alignSelf = $alignSelf;
+    }
+
+    if ($flexDirection) {
+      styles.flexDirection = $flexDirection;
+    }
+
+    if ($display) {
+      styles.display = $display;
+    }
+
+    if ($flex) {
+      styles.flex = $flex;
+    }
+
+    if ($justifyContent) {
+      styles.justifyContent = $justifyContent;
+    }
+
+    if ($justifySelf) {
+      styles.justifySelf = $justifySelf;
+    }
+
+    if ($position) {
+      styles.position = $position;
+    }
+
+    if ($width) {
+      styles.width = $width;
+    }
+
+    if ($minWidth) {
+      styles.minWidth = $minWidth;
+    }
+
+    if ($maxWidth) {
+      styles.maxWidth = $maxWidth;
+    }
+
+    if ($height) {
+      styles.height = $height;
+    }
+
+    if ($minHeight) {
+      styles.minHeight = $minHeight;
+    }
+
+    if ($maxHeight) {
+      styles.maxHeight = $maxHeight;
+    }
+
+    if ($overflow === 'scrollX') {
+      styles.overflowX = 'scroll';
+    } else if ($overflow === 'scrollY') {
+      styles.overflowY = 'scroll';
+    } else if ($overflow !== undefined) {
+      styles.overflow = $overflow;
+    }
+
+    if ($margin) {
+      styles.margin = sizing[$margin] || $margin;
+    }
+
+    if ($marginTop) {
+      styles.marginTop = sizing[$marginTop] || $marginTop;
+    }
+
+    if ($marginRight) {
+      styles.marginRight = sizing[$marginRight] || $marginRight;
+    }
+
+    if ($marginBottom) {
+      styles.marginBottom = sizing[$marginBottom] || $marginBottom;
+    }
+
+    if ($marginLeft) {
+      styles.marginLeft = sizing[$marginLeft] || $marginLeft;
+    }
+
+    if ($padding) {
+      styles.padding = sizing[$padding] || $padding;
+    }
+
+    if ($paddingTop) {
+      styles.paddingTop = sizing[$paddingTop] || $paddingTop;
+    }
+
+    if ($paddingRight) {
+      styles.paddingRight = sizing[$paddingRight] || $paddingRight;
+    }
+
+    if ($paddingBottom) {
+      styles.paddingBottom = sizing[$paddingBottom] || $paddingBottom;
+    }
+
+    if ($paddingLeft) {
```

Comment:
same with margins and padding - can be set to `0` and won't pass the condition

Reply:
I was actually checking `undefined` above, but stopped for some reason. I'm now doing that for all values.